pandas/core/groupby.py

```diff
@@ -929,6 +929,10 @@ def mean(self):
         Compute mean of groups, excluding missing values
 
         For multiple groupings, the result index will be a MultiIndex
+
+        Do not call this function via the numpy library. numpy's mean
```

Comment:
adding documentation about `just don't do this is silly` if you can fix them. as I said they should either work or fail with a nice error message.

you have to always alway reduce mental effort to use a library. that's the issue with numpy (and the reason we hiding it more and more), is that there is a lot of mental effort to understanding how to use it. pandas makes this much easier. if users don't ever know that numpy underlies pandas then that would be fine with me. all of this compat bizness is for a small class of users who DO want to know these things, so this is accomodation of THEM, not the general class of user.